lib/src/core/networking/client.dart

```diff
@@ -206,6 +207,55 @@ abstract class OpenAINetworkingClient {
     Map<String, dynamic>? body,
     http.Client? client,
   }) async {
+
+    var response = await postAndGetResponse(to: to, body: body, client: client);
+
+    final fileTypeHeader = "content-type";
+
+    final fileExtensionFromBodyResponseFormat =
+        response.headers[fileTypeHeader]?.split("/").last ?? "mp3";
+
+    final fileName =
+        outputFileName + "." + fileExtensionFromBodyResponseFormat;
+
+    File file = File(
+      "${outputDirectory != null ? outputDirectory.path : ''}" +
+          "/" +
+          fileName,
+    );
+
+    OpenAILogger.creatingFile(fileName);
+
+    await file.create();
+    OpenAILogger.fileCreatedSuccessfully(fileName);
+    OpenAILogger.writingFileContent(fileName);
+
+    file = await file.writeAsBytes(
+      response.bodyBytes,
+      mode: FileMode.write,
+    );
+
+    OpenAILogger.fileContentWrittenSuccessfully(fileName);
+
+    return onFileResponse(file);
+  }
+
+
+  static Future<Uint8List> postAndGetBytes({
+    required String to,
+    Map<String, dynamic>? body,
+    http.Client? client,
+  }) async {
+
+    var response = await postAndGetResponse(to: to, body: body, client: client);
+    return response.bodyBytes;
```

Comment:
in this case just get the bytes, should mean we can play with AudioPlayer `BytesSource` data source on platforms which don't support file operations